### PR TITLE
[operator] Add operator PoseQuaternionToHomogeneousMatrix

### PR DESCRIPTION
**Pull request by NoelieRamuzat**
A simple operation to transform a dg::vector containing a pose-quaternion (size 7: x, y, z, quat) to an Homogeneous Matrix.

### CODE REVIEW

File: src/matrix/operator.cpp [hunk 1]
```diff
@@ -356,6 +356,18 @@ struct MatrixHomoToPoseQuaternion
...
 REGISTER_UNARY_OP(MatrixHomoToPoseQuaternion, MatrixHomoToPoseQuaternion);
 
+struct PoseQuaternionToMatrixHomo
+    : public UnaryOpHeader<Vector, MatrixHomogeneous> {
+  void operator()(const dg::Vector &vect, MatrixHomogeneous &Mres) {
+    VectorQuaternion quat;
+    quat.w() = vect[6];
+    quat.vec() = vect.segment<3>(3);
+    Mres.translation() = vect.head<3>();
+    Mres.linear() = quat.toRotationMatrix();
```
Comment (jmirabel) [inline src/matrix/operator.cpp:366]: Minor comment: I prefer the shorter version below.
```suggestion
    Mres.translation() = vect.head<3>();
    Mres.linear() = VectorQuaternion(vect.tail<4>());
```

Comment (NoelieRamuzat) [reply src/matrix/operator.cpp:366]: Yes I totally agree !

Comment (NoelieRamuzat) [reply src/matrix/operator.cpp:366]: Just a comment, I think for the linear part it is:
`Mres.linear() = VectorQuaternion(vect.tail<4>()).toRotationMatrix();`

Comment (jmirabel) [reply src/matrix/operator.cpp:366]: There may be a `toRotationMatrix` missing.

**Comment by nim65s** [comment]
Hi @NoelieRamuzat, and thanks for this :)
Could you add a little unit test for this, in python and/or c++ ?

**Pull request merged**